src/script/conversation/ConversationRepository.ts

```diff
@@ -3922,23 +3941,21 @@ export class ConversationRepository {
    * User has received a welcome message in a conversation.
    *
    * @param conversationEntity Conversation entity user has received a welcome message in
-   * @param eventJson JSON data of 'conversation.mls-welcome' event
    * @returns Resolves when the event was handled
    */
-  private async onMLSWelcomeMessage(conversationEntity: Conversation, eventJson: ConversationMLSWelcomeEvent) {
+  private async onMLSWelcomeMessage(conversationEntity: Conversation) {
     // If we receive a welcome message in mls 1:1 conversation, we need to make sure proteus 1:1 is hidden (if it exists)
 
-    if (conversationEntity.type() !== CONVERSATION_TYPE.ONE_TO_ONE || !isMLSConversation(conversationEntity)) {
-      return;
-    }
-
-    const [otherUserId] = conversationEntity.participating_user_ids();
+    if (conversationEntity.type() === CONVERSATION_TYPE.ONE_TO_ONE && isMLSConversation(conversationEntity)) {
+      const [otherUserId] = conversationEntity.participating_user_ids();
 
-    if (!otherUserId) {
-      return;
+      if (otherUserId) {
+        await this.initMLS1to1Conversation(otherUserId, true);
+      }
     }
 
-    await this.initMLS1to1Conversation(otherUserId, true);
+    // After the welcome message is received (we were added to a conversation), we need to check mls conversation verification state
+    await this.mlsConversationVerificationStateHandler?.checkConversationVerificationState(conversationEntity);
```

Comment:
This is the actual fix - re-evaluate mls verification state every time we receive a welcome message